controllers/ibmpowervscluster_controller.go

```diff
@@ -111,6 +113,114 @@ func (r *IBMPowerVSClusterReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	return r.reconcile(clusterScope)
 }
 
+type updatePowerVSCluster struct {
+	cluster *infrav1beta2.IBMPowerVSCluster
+	mu      sync.Mutex
+}
+
+type reconcileResult struct {
+	reconcile.Result
+	error
+}
+
+func (update *updatePowerVSCluster) condition(condition bool, conditionArgs ...interface{}) {
+	update.mu.Lock()
+	defer update.mu.Unlock()
+	if condition {
+		conditions.MarkTrue(update.cluster, conditionArgs[0].(capiv1beta1.ConditionType))
+		return
+	}
+
+	conditions.MarkFalse(update.cluster, conditionArgs[0].(capiv1beta1.ConditionType), conditionArgs[1].(string), conditionArgs[2].(capiv1beta1.ConditionSeverity), conditionArgs[3].(string), conditionArgs[4:]...)
```

Comment:
If I pass it as a straight types, we would get a golint error for exceeding the max number of args which is 5. Also I would need to create a separate func for true and false condition. It would complicate things IMO.

>Should we add a check for verifying the min length to avoid index out of range?

Again, need to add a check on many places where we are updating the condition and won't be able to take major action if it errored. Ex [here](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/pull/1869/files#diff-11c67fcffb6910d7d88d6c399511174f09fbd5098e22e3aa37ee4f14f2302a8cR176) I want to return the main error instead of the error I m getting from this, it does n't look good to me. Scope of this is within the reconcile func and only two possible flow. I think the current way is ok IMO.